devito/targets/common/blocking.py

```diff
@@ -133,6 +133,9 @@ def fold_blockable_tree(iet, blockinner=True):
                 if len(i) >= 2:
                     groups.append(i)
         for i in groups:
+            # Performance heuristic: give up if fold length > 2:
+            if len(i) > 2:
+                continue
```

Comment:
Which heuristic?

Reply:
do not fold if there are too many loop nests to fold

don't worry to much about this code: this whole file will be completely dropped by `hoist-tiling` (WIP branch, you can already take a look), and so the entire concept of folding will disappear